src/Microsoft.IdentityModel.JsonWebTokens/Json/JsonClaimSet.cs

```diff
@@ -109,6 +109,10 @@ internal static void CreateClaimFromObject(List<Claim> claims, string claimType,
                     if (claim != null)
                         claims.Add(claim);
                 }
+            else
+            {
+                claims.Add(new Claim(claimType, value.ToString(), ClaimValueTypes.String, issuer, issuer));
```

Comment:
With the current code, when we read some unknown to IdentityModel claim, we cache it as a JSON string. With the new changes, the internal claims collection may now have custom classes and types other than what is checked above. So this makes sure these custom claims are added to ClaimsIdentity.